### PR TITLE
Support for subentities id (id_references)

### PR DESCRIPTION
**Pull request by Berardpi**
Example with an invoice:
When we receive an invoice from Connec!, we remember the Connec! line ids. The invoice is sent to the external application, and we then link the external app line ids received after creation with the one from Connec!, and we send the whole thing to Connec! so that it has the right id maps for the line ids.

The whole system require quite a bit of logic to retain only the relevant ids (I don't want to keep the whole connec! record in memory) and to merge everything at the end.

### CODE REVIEW

**Comment by Berardpi** [comment]
@alachaum @ouranos If one of you would have time for a nice code review :) I've tried to put as many comments as possible but I'd be happy to give you a tour of the code if needed.
(There's a lot of lines but it's mostly specs, no need to review them ;))

File: app/models/maestrano/connector/rails/concerns/connec_helper.rb [hunk 1]
```diff
@@ -93,23 +117,137 @@ def unfold_references_helper(entity, array_of_refs, organization)
...
-            unfold_references_helper(entity[ref], array_of_refs, organization)
+        return true if field.blank?
+        case field
+        when Array
+          bool = true
+          field.each do |f|
+            bool &= unfold_references_helper(f, array_of_refs.dup, organization)
+          end
+          bool
+        when HashWithIndifferentAccess
```
Comment (ouranos) [inline app/models/maestrano/connector/rails/concerns/connec_helper.rb]: `when Hash` ?

File: app/models/maestrano/connector/rails/concerns/connec_helper.rb [hunk 2]
```diff
@@ -93,23 +117,137 @@ def unfold_references_helper(entity, array_of_refs, organization)
...
+          unfold_references_helper(entity[ref], array_of_refs, organization)
+        end
+      end
+    end
+
+    # Transforms the references into an hash {record_references: [], id_references: []}
+    # References can either be an array (only record refenreces), or a hash
```
Comment (ouranos) [inline app/models/maestrano/connector/rails/concerns/connec_helper.rb]: typo

**Comment by ouranos** [comment]
I like the structure of the code: small methods and lot of comments but I have no idea what I'm looking at :grin: 
It's too abstract. I'm probably not the best to review this :wink:

**Comment by Berardpi** [comment]
@ouranos Thanks for the code review 🍺

**Pull request merged**